devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -0,0 +1,142 @@
+ *
+ */
+
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition (name: "ZigBee Accessory Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.remotecontroller", runLocally: false, minHubCoreVersion: '000.019.00012', executeCommandsLocally: false, mnmn: "SmartThings", vid: "generic-dimmer") {
+		capability "Actuator"
+		capability "Configuration"
+		capability "Switch"
+		capability "Button"
+		capability "Switch Level"
+
+		fingerprint profileId: "0104", inClusters: "0000,0003", outClusters: "0000,0004,0003,0006,0008,0005", manufacturer: "Aurora", model: "Remote50AU", deviceJoinName: "Aurora Wireless Wall Remote"
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+			tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute ("device.level", key: "SLIDER_CONTROL") {
+				attributeState "level", action:"switch level.setLevel"
+			}
+		}
+		main "switch"
+		details(["switch"])
+	}
+}
+
+def getSTEP() {10}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "description is $description"
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		if (event.name=="level" && event.value==0) {}
+		else {
+			sendEvent(event)
+		}
+	} else {
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		if (descMap && descMap.clusterInt == 0x0006) {
+			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
+				if (device.currentValue("level") == 0) {
+					sendEvent(name: "level", value: STEP)
+				}
+				sendEvent(name: "switch", value: device.currentValue("switch") == "on" ? "off" : "on")
+			} else {
+				log.warn "ON/OFF REPORTING CONFIG FAILED- error code:${cluster.data[0]}"
+			}
+		} else if (descMap && descMap.clusterInt == 0x0008) {
+			def currentLevel = device.currentValue("level") as Integer
+			if (descMap.commandInt == 0x02) {
+				def value = Math.min(currentLevel + STEP, 100)
+				log.debug "move to ${descMap.data}"
+				if (descMap.data[0] == "00") {
+					log.debug "move up"
+					sendEvent(name: "switch", value: "on")
+					sendEvent(name: "level", value: value)
+				} else if (descMap.data[0] == "01") {
+					log.debug "move down"
+					value = Math.max(currentLevel - STEP, 0)
+					sendEvent(name: "level", value: value)
```

Comment:
We usually don't create level events for a value of 0 because when they are turned on it is still 0.

Reply:
I did a little futzing about to make sure that if you turn it on from level = 0, level goes up to 10, but avoiding that situation in the first place is probably the better choice.